.github/workflows/ci.yml

```diff
@@ -86,10 +86,43 @@ jobs:
       - name: Install Rust
         uses: actions-rust-lang/setup-rust-toolchain@v1
         with:
-          toolchain: nightly-2023-12-28
+          toolchain: nightly-2025-01-10
           components: rust-src,rustfmt
+
+      # Rust Dependencies
```

Comment:
The duplication here feels pretty dirty, but maybe there's a better way for the jobs to depend on a separate job that builds the cache.